src/TropicalGeometry/hypersurface.jl

```diff
@@ -248,9 +248,9 @@ end
 
 Return the tropical polynomial used to construct `TropH`.  Raises an error if it is not cached.
 """
-function tropical_polynomial(TropH::TropicalHypersurface)
+function tropical_polynomial(TropH::TropicalHypersurface{minOrMax}) where minOrMax
     @req has_attribute(TropH,:tropical_polynomial) "no tropical polynomial cached"
-    return get_attribute(TropH,:tropical_polynomial)
+    return get_attribute(TropH,:tropical_polynomial)::Generic.MPoly{TropicalSemiringElem{minOrMax}}
```

Comment:
```suggestion
    return get_attribute(TropH,:tropical_polynomial)::mpoly_type(TropicalSemiringElem{minOrMax})
```
we have a nice getter for this